src/core/data.jl

```diff
@@ -57,3 +57,75 @@ function clean_status!(nw_data)
         end
     end
 end
+
+
+"Transforms a single network into a multinetwork with several deepcopies of the original network. Indexed from 0."
+function replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1)
+    return replicate_restoration_network(sn_data, count, Set(["baseMVA", "per_unit"]))
+end
+
+
+"Transforms a single network into a multinetwork with several deepcopies of the original network. Indexed from 0."
+function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int, global_keys::Set{String})
+    @assert count > 0
+    if _IMs.ismultinetwork(sn_data)
+        Memento.error(_PMs._LOGGER, "replicate_restoration_network can only be used on single networks")
+    end
+
+    name = get(sn_data, "name", "anonymous")
+
+    mn_data = Dict{String,Any}(
+        "nw" => Dict{String,Any}()
+    )
+
+    mn_data["multinetwork"] = true
+
+    sn_data_tmp = deepcopy(sn_data)
+    for k in global_keys
+        if haskey(sn_data_tmp, k)
+            mn_data[k] = sn_data_tmp[k]
+        end
+
+        # note this is robust to cases where k is not present in sn_data_tmp
+        delete!(sn_data_tmp, k)
+    end
+
+    mn_data["name"] = "$(count) period restoration of $(name)"
+
+    for n in 0:count
+        mn_data["nw"]["$n"] = deepcopy(sn_data_tmp)
+    end
+
+    for n in 0:count-1
+        if n != count
+            mn_data["nw"]["$n"]["time_elapsed"] = get(mn_data["nw"]["$n"],"time_elapsed",1)*calc_repair_time_elapsed(sn_data, n, count)
+        end
+    end
+
+    return mn_data
+end
+
+
+""
+function calc_repair_time_elapsed(data::Dict{String,<:Any}, nw::Int, count::Int)
```

Comment:
Let's move the logic of this function into `replicate_restoration_network`.  This function is dangerous because it uses global information (i.e. `cumulative_repairs`) to compute an incremental quantity i.e. repairs.  I would also add a field "repaired_total" to each network as a sanity check with the cardinality constraint.  There may be some challenges when `total_repairs/count` is not an integral number.

Reply:
Should this replication function still change time_elapsed or would that be bad practice? Should it only post a `:repair`,`cumulative_repairs` value to the mn data model?

This could be an issue for passing time_elapsed to the storage models and to the objective function in MLD.

Reply:
I decided to edit the time_elapsed inside of replicating the network.  Tests currently pass and I am reviewing the changes.